packages/salesforcedx-apex-debugger/src/adapter/apexDebug.ts

```diff
@@ -445,14 +880,286 @@ export class ApexDebug extends DebugSession {
     this.sendResponse(response);
   }
 
-  private logToDebugConsole(
+  protected async scopesRequest(
+    response: DebugProtocol.ScopesResponse,
+    args: DebugProtocol.ScopesArguments
+  ): Promise<void> {
+    const frameInfo = this.stackFrameInfos.get(args.frameId);
+    if (!frameInfo) {
+      this.sendErrorResponse(
+        response,
+        2020,
```

Comment:
Did you find out what this does and if we need it?

Reply:
We can replace the code with a message object `codeOrMessage: number | DebugProtocol.Message`. Would you prefer a message object?

Reply:
Actually, I'll replace with a log message and simply return an empty result here.

Reply:
In general if we have a type of the form `TypeAOrTypeB` that means we should be explicit about it as being a _disjoint_ union rather than just a union. I would try to avoid them. But, if they are necessary, we should use an explicit `Either` like what Scala does. See https://alvinalexander.com/scala/scala-either-left-right-example-option-some-none-null

Is there a use case where we should introduce the generic Either<T> type into our code base? If there is, we should instead of propagating a bunch of `TypeAOrTypeB` types.